content/en/continuous_integration/guides/pipeline_data_model.md

```diff
@@ -0,0 +1,88 @@
+---
+title: Pipeline Data Model And Execution Types 
+kind: guide
+description: Learn how Pipelines are modeled and what execution types are supported by CI Visibility
+---
+
+## Data model
+
+{{< site-region region="gov" >}}
+<div class="alert alert-warning">CI Visibility is not available in the selected site ({{< region-param key="dd_site_name" >}}) at this time.</div>
+{{< /site-region >}}
+
+Pipeline executions are modeled as traces, similar to an [APM distributed trace][1], where spans represent the execution of different parts of the pipeline. The CI Visibility data model for representing pipeline executions consists of four levels:
+
+| Level Name | Description |
+| ---------- | ----------- |
+| Pipeline   | The top-level root span that contains all other levels as children. It represents the overall execution of a pipeline from start to finish. This level is sometimes called build or workflow in some CI providers. |
+| Stage      | Serves as a grouping of jobs under a user-defined name. Many CI providers do not have this level. |
+| Job        | The smallest unit of work where commands are executed. All tasks at this level should be performed on a single node. |
+| Step       | In some CI providers, this level represents a shell script or an action executed within a job. |
+
+When a pipeline, stage, job, or step finishes, its execution data should be sent to Datadog. See the list of [supported CI providers][2] to set up your pipeline visibility. If your CI provider or workflow is not supported, you can use the [public API endpoint][3] to send your pipeline executions to CI Visibility.
+
+{{< img src="ci/ci-pipeline-execution.png" alt="Example of a pipeline execution trace" style="width:100%;">}}
+
+### Pipeline unique IDs
+
+All pipeline executions within a level must have a unique identifier. Providing repeated IDs result in undefined behavior in CI Visibility.
+
+## Pipeline execution types
+
+### Normal run
+
+The normal run of a pipeline execution follows the flow depicted below:
+
+{{< img src="ci/ci-pipeline-normal-execution-flow.png" alt="Depiction of a normal pipeline execution" style="width:100%;">}}
+
+**Note**: depending on the provider, some levels may be missing. For example, stages might not exist, and jobs may run in parallel or sequence, or a combination of both.
+
+After the completion of each component, a payload must be sent to Datadog with all the necessary data to represent the execution. Datadog will process this data, store it as pipeline events, and display it as traces in the CI Visibility product.
```

Comment:
```suggestion
After the completion of each component, a payload must be sent to Datadog with all the necessary data to represent the execution. Datadog will process this data and display it in the CI Visibility product.
```
I'm not sure what storing something as pipeline events means. Here we are talking about also stages, jobs, etc. which are not really pipeline events

Reply:
Pipeline event, meaning an event from the CIPIPELINE track. This term in commonly used in our APIs. For example https://docs.datadoghq.com/api/latest/ci-visibility-pipelines/#get-a-list-of-pipelines-events